### ISSUE TITLE
JavaScript exception on first launch of RStudio

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#2963]
### System details

    RStudio Edition : Desktop
    RStudio Version : v1.2.715
    OS Version      : macOS 10.13.5
    R Version       : 3.5.0

### Steps to reproduce the problem

Remove `~/.rstudio-desktop` just to clear any client state, then attempt to launch RStudio from the terminal.

### Describe the problem in detail

A JavaScript exception is emitted:

```
$ /Applications/RStudio.app/Contents/MacOS/RStudio
"WARNING: Exception occured dispatching event: ui_prefs_changed"
"Exception: com.google.gwt.event.shared.UmbrellaException: Exception caught: (TypeError) : Cannot read property 'background' of undefined"
"Error: com.google.gwt.event.shared.UmbrellaException: Exception caught: (TypeError) : Cannot read property 'background' of undefined\n    at UmbrellaException_5_g$.createError_0_g$ [as createError_2_g$] (rstudio-0.js:12374:10)\n    at UmbrellaException_5_g$.initializeBackingError_0_g$ [as initializeBackingError_1_g$] (rstudio-0.js:12438:48)\n    at UmbrellaException_5_g$.Throwable_4_g$ (rstudio-0.js:12281:8)\n    at UmbrellaException_5_g$.Exception_4_g$ (rstudio-0.js:12551:18)\n    at UmbrellaException_5_g$.RuntimeException_4_g$ (rstudio-0.js:12606:18)\n    at UmbrellaException_5_g$.UmbrellaException_3_g$ (rstudio-0.js:32797:25)\n    at new UmbrellaException_5_g$ (rstudio-0.js:32868:26)\n    at EventBus_3_g$.fireEvent_2_g$ [as fireEvent_41_g$] (rstudio-0.js:32360:23)\n    at EventBus_3_g$.fireEvent_15_g$ [as fireEvent_44_g$] (rstudio-0.js:166602:49)\n    at EventBus_3_g$.fireEvent_14_g$ [as fireEvent_41_g$] (rstudio-0.js:166580:8)\n    at ClientEventDispatcher_1_g$.dispatchEvent_9_g$ [as dispatchEvent_14_g$] (rstudio-0.js:260719:28)\n    at ClientEventDispatcher$1_1_g$.execute_480_g$ [as execute_1366_g$] (rstudio-0.js:261150:26)\n    at $executeRepeating_0_g$ (rstudio-0.js:15008:47)\n    at runScheduledTasks_0_g$ (rstudio-0.js:14701:13)\n    at SchedulerImpl_1_g$.flushPostEventPumpCommands_0_g$ [as flushPostEventPumpCommands_1_g$] (rstudio-0.js:14805:5)\n    at SchedulerImpl$Flusher_1_g$.execute_6_g$ [as execute_1366_g$] (rstudio-0.js:14959:23)\n    at execute_5_g$ (rstudio-0.js:14672:19)\n    at apply_0_g$ (rstudio-0.js:14390:28)\n    at entry0_0_g$ (rstudio-0.js:14449:16)\n    at rstudio-0.js:14428:14\n    at callback_0_g$ (rstudio-0.js:14726:45)"
"Caused by: com.google.gwt.core.client.JavaScriptException: (TypeError) : Cannot read property 'background' of undefined"
"TypeError: Cannot read property 'background' of undefined\n    at getEditorThemeProperty_0_g$ (rstudio-0.js:171581:33)\n    at getBackground_0_g$ (rstudio-0.js:171574:10)\n    at HtmlMessageListener$1_1_g$.onUiPrefsChanged_0_g$ [as onUiPrefsChanged_4_g$] (rstudio-0.js:102638:22)\n    at UiPrefsChangedEvent_2_g$.dispatch_461_g$ [as dispatch_1064_g$] (rstudio-0.js:296678:16)\n    at UiPrefsChangedEvent_2_g$.dispatch_460_g$ [as dispatch_833_g$] (rstudio-0.js:296673:8)\n    at UiPrefsChangedEvent_2_g$.dispatch_0_g$ [as dispatch_832_g$] (rstudio-0.js:29685:8)\n    at dispatchEvent_3_g$ (rstudio-0.js:32412:14)\n    at HandlerManager$Bus_1_g$.doFire_0_g$ [as doFire_1_g$] (rstudio-0.js:32522:9)\n    at HandlerManager$Bus_1_g$.fireEvent_3_g$ [as fireEvent_42_g$] (rstudio-0.js:32601:8)\n    at EventBus_3_g$.fireEvent_2_g$ [as fireEvent_41_g$] (rstudio-0.js:32354:24)\n    at EventBus_3_g$.fireEvent_15_g$ [as fireEvent_44_g$] (rstudio-0.js:166602:49)\n    at EventBus_3_g$.fireEvent_14_g$ [as fireEvent_41_g$] (rstudio-0.js:166580:8)\n    at ClientEventDispatcher_1_g$.dispatchEvent_9_g$ [as dispatchEvent_14_g$] (rstudio-0.js:260719:28)\n    at ClientEventDispatcher$1_1_g$.execute_480_g$ [as execute_1366_g$] (rstudio-0.js:261150:26)\n    at $executeRepeating_0_g$ (rstudio-0.js:15008:47)\n    at runScheduledTasks_0_g$ (rstudio-0.js:14701:13)\n    at SchedulerImpl_1_g$.flushPostEventPumpCommands_0_g$ [as flushPostEventPumpCommands_1_g$] (rstudio-0.js:14805:5)\n    at SchedulerImpl$Flusher_1_g$.execute_6_g$ [as execute_1366_g$] (rstudio-0.js:14959:23)\n    at execute_5_g$ (rstudio-0.js:14672:19)\n    at apply_0_g$ (rstudio-0.js:14390:28)\n    at entry0_0_g$ (rstudio-0.js:14449:16)\n    at rstudio-0.js:14428:14\n    at callback_0_g$ (rstudio-0.js:14726:45)"
"console.groupEnd"
"console.groupEnd"
```

### Describe the behavior you expected

No exception.

**Comment by jmcphers** [comment]
I think this is because we assume here that the theme name will match one from the hard-coded list:

https://github.com/rstudio/rstudio/blob/dcf51c26a14b929ab26d5f1136320ec819f19f20/src/gwt/src/org/rstudio/studio/client/application/ui/RStudioThemes.java#L189

@MariaSemple I'm guessing your upcoming changes will obviate this. Is this correct?

**Comment by MariaSemple** [comment]
Yes I think I saw this happening when the theme name being requested wasn't in the list returned to the server (after I removed the hard coded list). It should be fixed with my next PR.

**Comment by jmcphers** [comment]
Should be fixed by https://github.com/rstudio/rstudio/commit/830f88fc7b5a8d4b5bfdc79ced3b28442f348970.

**Comment by dfalty** [comment]
Verified fixed on version 1.2.1090.
